src/box/errcode.h

```diff
@@ -141,287 +141,313 @@ struct errcode_record {
 	_(ER_MISSING_SNAPSHOT, 93,		"Can't find snapshot") \
-	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index in space '%s'") \
-	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow when performing '%c' operation on field %s") \
+	_(ER_CANT_UPDATE_PRIMARY_KEY, 94,	"Attempt to modify a tuple field which is part of primary index in space '%s'", "space", STRING) \
+	_(ER_UPDATE_INTEGER_OVERFLOW, 95,	"Integer overflow when performing '%c' operation on field %s", "operation", CHAR, "field", STRING) \
 	_(ER_GUEST_USER_PASSWORD, 96,		"Setting password for guest user has no effect") \
 	_(ER_TRANSACTION_CONFLICT, 97,		"Transaction has been aborted by conflict") \
-	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported %s privilege '%s'") \
-	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function '%s': %s") \
-	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported language '%s' specified for function '%s'") \
-	_(ER_RTREE_RECT, 101,			"RTree: %s must be an array with %u (point) or %u (rectangle/box) numeric coordinates") \
+	_(ER_UNSUPPORTED_PRIV, 98,		"Unsupported %s privilege '%s'", "object_type", STRING, "privilege", STRING) \
+	_(ER_LOAD_FUNCTION, 99,			"Failed to dynamically load function '%s': %s", "func", STRING, "details", STRING) \
+	_(ER_FUNCTION_LANGUAGE, 100,		"Unsupported language '%s' specified for function '%s'", "value", STRING, "func", STRING) \
+	_(ER_RTREE_RECT, 101,			"RTree: %s must be an array with %u (point) or %u (rectangle/box) numeric coordinates", \
+						"", STRING, "", UINT, "", UINT) \
 	_(ER_PROC_C, 102,			"%s") \
-	_(ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, "Unknown RTREE index distance type %s") \
-	_(ER_PROTOCOL, 104,			"%s") \
-	_(ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	"Space %s has a unique secondary index and does not support UPSERT") \
-	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space: got {%s}, expected {%s}") \
-	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part %u: %s") \
-	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options: %s") \
-	_(ER_WRONG_SCHEMA_VERSION, 109,		"Wrong schema version, current: %d, in request: %llu") \
-	_(ER_MEMTX_MAX_TUPLE_SIZE, 110,		"Failed to allocate %u bytes for tuple: tuple is too large. Check 'memtx_max_tuple_size' configuration option.") \
-	_(ER_WRONG_SPACE_OPTIONS, 111,		"Wrong space options: %s") \
+	/* ER_UNKNOWN_RTREE_INDEX_DISTANCE_TYPE, 103, Unused */ \
+	_(ER_PROTOCOL, 104,			"%s", "details", STRING) \
+	/* ER_UPSERT_UNIQUE_SECONDARY_KEY, 105,	Unused */ \
+	_(ER_WRONG_INDEX_RECORD, 106,		"Wrong record in _index space: %s", "details", STRING) \
+	_(ER_WRONG_INDEX_PARTS, 107,		"Wrong index part %u: %s", "partno", UINT, "details", STRING) \
+	_(ER_WRONG_INDEX_OPTIONS, 108,		"Wrong index options: %s", "details", STRING) \
+	_(ER_WRONG_SCHEMA_VERSION, 109,		"Wrong schema version, current: %d, in request: %llu", "actual", ULLONG, "expected", ULLONG) \
```

Comment:
I think you should swap "expected" and "actual".

Reply:
Discussed f2f and decided it is correct (actually the swap should be in the error message).